### DISCUSSION TITLE
Pdf splitting/stacking

### DISCUSSION THREAD

**Discussion by PasaOpasen** [pypdfium2-team/pypdfium2#170] [Q&A] [answered]
Is it possible to split pdf per pages or stack several pdf files to one?

**Comment by mara004** [comment, marked as answer]
Yes. You can create new pdfs and import pages from foreign pdfs. Thereby you can split/merge pdfs and move pages programmatically.

However, `main` doesn't have a helper method for that yet, so I suggest that you take a look at the `devel-new` branch, which provides `PdfDocument.import_pages()`:
https://github.com/pypdfium2-team/pypdfium2/blob/5c05d6f5f9d6a1e1a9d01e674e5fb247f39463a6/src/pypdfium2/_helpers/document.py#L366-L394

So basically what you would do is something like this
```python
# example: split a 6-page document after page 3

import pypdfium2 as pdfium

src_pdf = pdfium.PdfDocument("/abspath/to/source.pdf")

out_a = pdfium.PdfDocument.new()
out_b = pdfium.PdfDocument.new()

out_a.import_pages(src_pdf, [0, 1, 2])
out_b.import_pages(src_pdf, [3, 4, 5])

# (note: saving to paths directly needs devel-new as well, main only accepts buffers)
out_a.save("/abspath/to/out_a.pdf")
out_b.save("/abspath/to/out_b.pdf")
```

**Reply by PasaOpasen** [reply]
Thank U!

**Reply by PasaOpasen** [reply]
@mara004 but this version is not contained in current release, will be with next?

**Reply by mara004** [reply]
No, unfortunately not yet. There's trouble around test coverage in devel-new that I'd like to address before merging.
And I've also got some serious personal problems that prevent me from working on it.

Anyway, you may be able to pin your setup to devel-new if you like.

**Reply by mara004** [reply]
@PasaOpasen I've merged the branch, so the next version will include the above features. There's a beta release already.
